internal/fsutil/create.go

```diff
@@ -18,8 +21,29 @@ type CreateOptions struct {
 	MakeParents bool
 }
 
+type Info struct {
+	Path string
+	Mode fs.FileMode
+	Hash string
+	Size uint
+	Link string
+}
+
+type Creator struct {
+	// Created keeps track of information about the files created. If a file
+	// is created several times it only tracks the latest one.
+	Created map[string]Info
+}
+
+func NewCreator() *Creator {
+	return &Creator{Created: make(map[string]Info)}
+}
+
 // Creates a filesystem entry according to the provided options.
-func Create(o *CreateOptions) error {
+func (c Creator) Create(o *CreateOptions) error {
+	rp := readerProxy{inner: o.Data, h: sha256.New()}
+	o.Data = &rp
```

Comment:
This doesn't seem right, as it's mutating an options value from the call site in an unexpected way (for the caller).

Why is it needed?

Reply:
Changed